extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -8,45 +8,38 @@ import * as vscode from 'vscode';
 import * as path from 'path';
 import * as fs from 'fs';
 import * as yaml from 'js-yaml';
+import * as fg from 'fast-glob';
 import { BookTreeItem, BookTreeItemType } from './bookTreeItem';
+import { maxBookSearchDepth, notebookConfigKey } from '../common/constants';
 import * as nls from 'vscode-nls';
 const localize = nls.loadMessageBundle();
 
-
 export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeItem>, azdata.nb.NavigationProvider {
 	readonly providerId: string = 'BookNavigator';
 
 	private _onDidChangeTreeData: vscode.EventEmitter<BookTreeItem | undefined> = new vscode.EventEmitter<BookTreeItem | undefined>();
 	readonly onDidChangeTreeData: vscode.Event<BookTreeItem | undefined> = this._onDidChangeTreeData.event;
-	private _tableOfContentsPath: string[];
+	private _tableOfContentPaths: string[] = [];
 	private _allNotebooks = new Map<string, BookTreeItem>();
 	private _extensionContext: vscode.ExtensionContext;
 	private _throttleTimer: any;
 	private _resource: string;
 
 	constructor(workspaceFolders: vscode.WorkspaceFolder[], extensionContext: vscode.ExtensionContext) {
-		let workspacePaths: string[] = workspaceFolders.map(a => a.uri.fsPath);
-		this._tableOfContentsPath = this.getTableOfContentFiles(workspacePaths);
-		let bookOpened: boolean = this._tableOfContentsPath && this._tableOfContentsPath.length > 0;
-		vscode.commands.executeCommand('setContext', 'bookOpened', bookOpened);
+		this.getTableOfContentFiles(workspaceFolders).then(() => undefined, (err) => { console.log(err); });
 		this._extensionContext = extensionContext;
 	}
 
-	private getTableOfContentFiles(directories: string[]): string[] {
-		let tableOfContentPaths: string[] = [];
-		let paths: string[];
-		directories.forEach(dir => {
-			paths = fs.readdirSync(dir);
-			paths.forEach(filename => {
-				let fullPath = path.join(dir, filename);
-				if (fs.statSync(fullPath).isDirectory()) {
-					tableOfContentPaths = tableOfContentPaths.concat(this.getTableOfContentFiles([fullPath]));
-				} else if (filename === 'toc.yml') {
-					tableOfContentPaths.push(fullPath);
-				}
-			});
-		});
-		return tableOfContentPaths;
+	async getTableOfContentFiles(workspaceFolders: vscode.WorkspaceFolder[]): Promise<void> {
+		let notebookConfig = vscode.workspace.getConfiguration(notebookConfigKey);
+		let maxDepth = notebookConfig[maxBookSearchDepth];
```

Comment:
Questions:
- What if a user enters a negative number?
- What if a user enters 0 for this?
- What if user makes it undefined?

We should have some basic handling to fall back to a value in those cases. Usually 0 indicates "infinite depth" for example.

Reply:
I will use the default value (5) if the user enters a negative number or undefined, and use 0 for infinite depth.